lib/mySociety/Deploy.pm

```diff
@@ -52,25 +52,13 @@ sub setup_conf {
     if (exists($conf->{git_repository})) {
         die "must specify 'git_user' in vhost config" if !exists($conf->{git_user});
         die "can't have blank 'git_user', use 'anon' for anonymous" if $conf->{git_user} eq '';
-        if (exists($conf->{git_ref_server_mapping})) {
-            die "git_ref_server_mapping defined, but no mapping for $hostname"
-                if !exists($conf->{git_ref_server_mapping}{$hostname});
-            $conf->{git_ref} = $conf->{git_ref_server_mapping}{$hostname};
-        } elsif (!exists($conf->{git_ref})) {
-            $conf->{git_ref} = 'origin/master';
-        }
+        $conf->{git_ref} = 'origin/master' if !exists($conf->{git_ref});
         $vcspath = $conf->{git_repository};
     }
 
     if (exists($conf->{private_git_repository})) {
-        if (exists($conf->{private_git_ref_server_mapping})) {
-            die "private_git_ref_server_mapping defined, but no mapping for $hostname"
-                if !exists($conf->{private_git_ref_server_mapping}{$hostname});
-            $conf->{private_git_ref} = $conf->{private_git_ref_server_mapping}{$hostname};
-        } elsif (!exists($conf->{private_git_ref})) {
-            $conf->{private_git_ref} = 'origin/master' if !exists($conf->{private_git_ref});
-        }
-        $vcspath = $conf->{private_git_repository} if !$vcspath;
+        $conf->{private_git_ref} = 'origin/master' if !exists($conf->{private_git_ref});
+        $vcspath = $conf->{private_git_repository};
```

Comment:
This line changes behaviour, not just removing ...server_mapping - previously it was only setting vcspath if it hadn't been set by git_repository above, now it is always being set.

Reply:
Yes, true; I think I assumed that we'd only ever have one or the other so the conditional seemed unnecessary  (which is the case for current services), but I see that there's nothing anywhere to prevent us having both except that I think `make_git_repository` on timestamped deploys would end up trying to clone both into the same places.

Happy to revert this to conditional for consistency and for the purposes of this commit, but unless I'm wrong above probably need to do something about that as well at some point.